wake/build.wake

```diff
@@ -34,28 +34,43 @@ global def u500ScalaModule =
 global def u500VC707DUTPlan =
   def name = "u500VC707DUTPlan"
   def testharness = "sifive.fpgashells.shell.xilinx.VC707Shell"
-  def configs =
-    "sifive.freedom.unleashed.DevKitU500FPGADesign",
-    "sifive.freedom.unleashed.WithDevKit50MHz",
-    Nil
-  makeRocketChipDUTPlan name u500ScalaModule testharness ""
-  | setRocketChipDUTPlanConfigs configs
+  def config = "sifive.freedom.unleashed.DevKitU500FPGADesign"
+  def frequency = 50
+  def scalaModule = u500ScalaModule
+  makeRocketChipDUTPlan name scalaModule testharness config
+  | setRocketChipDUTPlanTestharness testharness
+  | addFPGADevKitFrequencyConfig frequency
 
 global def u500VCU118DUTPlan =
   def name = "u500VCU118DUTPlan"
   def testharness = "sifive.fpgashells.shell.xilinx.VCU118Shell"
-  def configs =
-    "sifive.freedom.unleashed.DevKitU500FPGADesign",
-    "sifive.freedom.unleashed.WithDevKit50MHz",
-    Nil
-  makeRocketChipDUTPlan name u500ScalaModule testharness ""
-  | setRocketChipDUTPlanConfigs configs
+  def config = "sifive.freedom.unleashed.DevKitU500FPGADesign"
+  def frequency = 50
+  def scalaModule = u500ScalaModule
+  makeRocketChipDUTPlan name scalaModule testharness config
+  | setRocketChipDUTPlanTestharness testharness
+  | addFPGADevKitFrequencyConfig frequency
 
 global def e300ArtyDUTPlan =
   def name = "e300ArtyDUTPlan"
   def testharness = "sifive.freedom.everywhere.e300artydevkit.E300ArtyDevKitFPGAChip"
-  def configs =
-    "sifive.freedom.everywhere.e300artydevkit.E300ArtyDevKitConfig",
-    Nil
-  makeRocketChipDUTPlan name e300ScalaModule testharness ""
-  | setRocketChipDUTPlanConfigs configs
+  def config = "sifive.freedom.everywhere.e300artydevkit.E300ArtyDevKitConfig"
+  makeRocketChipDUTPlan name e300ScalaModule testharness config
+
+def devKitFrequencyConfig = match _
+  25  = "sifive.freedom.unleashed.WithDevKit25MHz"
+  50  = "sifive.freedom.unleashed.WithDevKit50MHz"
+  100 = "sifive.freedom.unleashed.WithDevKit100MHz"
+  125 = "sifive.freedom.unleashed.WithDevKit125MHz"
+  150 = "sifive.freedom.unleashed.WithDevKit150MHz"
+  200 = "sifive.freedom.unleashed.WithDevKit200MHz"
+  frequency =
+    def _ = printLevel logError "FPGADevKit: {str frequency} is not a valid frequency"
+    "INVALID.FREQUENCY.CONFIG"
+
+global def addFPGADevKitFrequencyConfig frequency plan =
```

Comment:
Could you add a comment here describing what this function does? I'd like for us to get into the habit of better documenting what global functions do. I think especially documenting what the arguments are here would make it easier to use this function, since it's not always clear what specific type the `plan` is.